src/components/view/Navigation/index.js

```diff
@@ -34,47 +30,58 @@ class Navigation extends React.Component {
 
     constructor(props) {
         super(props);
-        this.navigateToRoute = this.navigateToRoute.bind(this);
+        this.handleLogout = this.handleLogout.bind(this);
+        this.handleToggle = this.handleToggle.bind(this);
         this.themePalette = this.props.muiTheme.palette;
+        this.state = { drawerOpen: false };
+    }
+
+    handleToggle () {
+        // doesn't show nevigation unless logged in.
+        const { user } = this.props;
+        if (user.uid && user.uid.length > 0) {
+            this.setState({drawerOpen: !this.state.drawerOpen});
+        }
+    }
+
+    getMenuItemHandler (routeName) {
+        return function () {
+            this.setState({drawerOpen: false});
+            this.props.history.push(routeName);
+        }.bind(this);
     }
-    
-    buildIconMenu (actions) {
-        const { logout } = actions;
-        return (
-            <IconMenu
-                iconButtonElement={<IconButton><ExpandMoreIcon /></IconButton>}
-                anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
-                targetOrigin={{ horizontal: 'left', vertical: 'top' }}
-            >
-                <MenuItem onTouchTap={logout} primaryText="Logout" />
-            </IconMenu>
-        );
-    };
 
-    navigateToRoute (someArg, evt, tabEl) {
-        // close any open event item
-        // this.toggleEventDetail();
-        // navigate
-        this.props.history.push(tabEl.props['data-route']);
-    };
+    handleLogout () {
+        this.setState({drawerOpen: false});
+        this.props.logout();
+    }
 
     render () {
-        const { user, logout } = this.props;
-        
-        const iconMenu = this.buildIconMenu({ logout });
+        const { user } = this.props;
+
+        const avatarSize = 60,
+            paddingSize = 15,
+            drawerWidth = Math.min(450, window.outerWidth * .8);
```

Comment:
I know I came up with this Math.min idea yesterday, but I'm curious if we could use the flexbox grid's classes here?  

e.g. `<Drawer className="col-xs-10 col-sm-8 col-md-4 col-lg-2 col-xl-1" ...`

What you have implemented certainly works and is great for an MVP (and I will accept it as is), but would love to spend a few minutes thinking about how this might behave if the window is resized after loading and if there are any clever solutions for addressing that without putting window resize listeners everywhere. Maybe that just ends up being a separate story.

#consider #optional